src/main/java/seedu/kitchenhelper/storage/Storage.java

```diff
@@ -1,5 +1,152 @@
+ */
 public class Storage {
 
-}
+    private String filePath;
+
+    /**
+     * Constructor for Storage.
+     * @param filePath String of filepath for stored data.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Gets the saved Ingredient data from text file.
+     * @return ArrayList Contains data from saved text file
+     * @throws FileNotFoundException If file from file path does not exists.
+     */
+    public ArrayList<Ingredient> getIngredientData() throws FileNotFoundException {
+        ArrayList<Ingredient> ingredientList = new ArrayList<>();
+        File file = new File(filePath);
+        Scanner scanner = new Scanner(file);
+
+        while (scanner.hasNext()) {
+
+            String userData = scanner.nextLine();
+            String[] getName = userData.split("/n ");
+            String[] getCat = getName[1].split(" /c ");
+            String[] getQuantity = getCat[1].split(" /q ");
+            String[] getPrice = getQuantity[1].split(" /p ");
+            String[] getExpiry = getPrice[1].split(" /e ");
+
+            String name = getCat[0];
+            String category =  getQuantity[0];
+            Integer quantity = Integer.parseInt(getPrice[0]);
+            Double price = Double.parseDouble(getExpiry[0]);
+            String expiry = getExpiry[1];
+
+            switch (category) {
+            case "Dairy":
+                Ingredient diary = new Dairy(name, category, quantity, price, expiry);
+                ingredientList.add(diary);
+                break;
+            case "Drink": {
+                Ingredient drink = new Drink(name, category, quantity, price, expiry);
+                ingredientList.add(drink);
+                break;
+            }
+            case "Fruit": {
+                Ingredient fruit = new Fruit(name, category, quantity, price, expiry);
+                ingredientList.add(fruit);
+                ;
+                break;
+            }
+            case "Meat": {
+                Ingredient meat = new Meat(name, category, quantity, price, expiry);
+                ingredientList.add(meat);
+                break;
+            }
+            case "Miscellaneous": {
+                Ingredient miscellaneous = new Miscellaneous(name, category, quantity, price, expiry);
+                ingredientList.add(miscellaneous);
+                break;
+            }
+            case "Staple": {
+                Ingredient staple = new Staple(name, category, quantity, price, expiry);
+                ingredientList.add(staple);
+                break;
+            }
+            case "Vegetable": {
+                Ingredient vegetable = new Vegetable(name, category, quantity, price, expiry);
+                ingredientList.add(vegetable);
+                break;
+            }
+            default:
+                throw new IllegalStateException("Unexpected value: " + category);
+            }
```

Comment:
A function should be doing one thing but there's two here (parsing the ingredients with its content & adding), it would be better if there was another function (`loadingIngredients()`?) to call the switching of ingredient's category. But all is good! 😄